### PR TITLE
49 minimum marketable product   add password functionality

### PR DESCRIPTION
**Pull request by RadiationOcelot**
- Added password hashing functionality.
- Passwords are now stored in the database hashed and salted
- Passwords are checked when logging in
- Current User table schema has changed due to now having to store a randomly generated password salt
- `SECRET_KEY` is now an environmental variable and not a placeholder string

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -5,14 +5,26 @@
...
     # Integer column type with primary_key=True will enable auto-increment
     id = db.Column(db.Integer, primary_key=True)
     username = db.Column(db.Text, nullable=False, unique=True)
     points = db.Column(db.Integer, nullable=False)
-    # I believe there are more secure ways of implementing this
+    # Maybe we should use something like Character rather than Text to limit the length of hash stored?
+    # Currently cannot think of a substantial reason to do this
     password_hash = db.Column(db.Text, nullable=False)
+    # Same thing for the salt
+    password_salt = db.Column(db.Text, nullable=False)
```
Comment (RedBlueCarrots) [inline app/models.py:19]: We should alter this in a DB migration

**Review by RedBlueCarrots — CHANGES_REQUESTED** [review]
Looks good to me, except changes to database schema should be done in a db migration. Maybe do #51 while you are at it as well>

**Comment by RadiationOcelot** [comment]
Altered the database via a Flask-SQL-Alchemy migration.

How to do a migration:

1.  Change the database in some way (adding new rows, altering columns etc.)
2. In the WikiPath folder, run `flask db migrate -m "useful_migration_name"`
3. In migrations/versions, find the python script correlating to your migration
4. Very Important! Check the generated script and confirm it's doing the changes that you want to do
5. Back in the WikiPath folder, run `flask db upgrade`

If you simply want to upgrade an existing database you have in local storage with migration scripts made by others, run `flask db upgrade`.

File: migrations/README [hunk 2]
```diff
@@ -0,0 +1 @@
+Single-database configuration for Flask.
```
Comment (DeltaO3) [inline migrations/README:1]: Do we need to edit this README in anyway / is it contributing anything?

File: migrations/versions/f16b32004a3d_wikiaura_addition.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+# revision identifiers, used by Alembic.
+revision = 'f16b32004a3d'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.alter_column('user', 'points', new_column_name = 'WikiAura')
+
```
Comment (DeltaO3) [inline migrations/versions/f16b32004a3d_wikiaura_addition.py:21]: Shouldnt the migration add the password salt column as well?

**Review by DeltaO3 — CHANGES_REQUESTED** [review]
Looks pretty good (and im assuming most of the stuff in the migrations folder was generated/ is mostly correct for now), but shouldnt you also add the new salt column to the database in the first migration?

**Review by DeltaO3 — APPROVED** [review]
Salt added to migrations is good! LGTM :)

**Pull request merged**